uk.co.compucorp.civicrm.hrcore/CRM/HRCore/CMSData/CMSRoleServiceFactory.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+use CRM_HRCore_CMSData_Role_DrupalRoleService as DrupalRoleService;
+use CRM_HRCore_CMSData_Role_RoleServiceInterface as RoleServiceInterface;
+
+class CRM_HRCore_Service_CMSRoleServiceFactory {
+
+  /**
+   * @return RoleServiceInterface
+   */
+  public static function create() {
+    $userSystem = CRM_Core_Config::singleton()->userSystem;
+
+    switch (get_class($userSystem)) {
+      case CRM_Utils_System_Drupal::class:
+        return new DrupalRoleService();
+      default:
+        $msg = sprintf('Unrecognized system "%s"', get_class($userSystem));
```

Comment:
Wondering if the user framework `CRM_Core_Config::singleton()->userFramework` won't be better to display here in case there is an unrecognised system.  The return value seems more user friendly and less technical than returning the class name.

Reply:
I didn't know about `userFramework`, this is much better including to use in the switch. I've changed this and the other place where I did a similar comparison (`CRM_HRCore_CMSData_SiteInformationFactory`)